code/game/objects/structures/tables_racks.dm

```diff
@@ -328,22 +328,53 @@
 	if(user.stat || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || !Adjacent(user) || !can_be_flipped || is_ventcrawling(user))
 		return
 
+	var/flip_speed = get_flip_speed(user)
+
 	if(!flipped)
+
+		if(flip_speed > 0)
+			user.visible_message("<span class='warning'[user] starts trying to flip [src]!</span>", "<span class='warning'>You start trying to flip [src][flip_speed >= 5 SECONDS ? " (it'll take about [flip_speed / 10] seconds)." : ""].</span>")
+			if(!do_after(user, flip_speed, TRUE, src))
+				user.visible_message("<span class='notice'[user] gives up on trying to flip [src].</span>")
+				return
 		if(!flip(get_cardinal_dir(user, src)))
 			to_chat(user, "<span class='notice'>It won't budge.</span>")
 			return
 
-		user.visible_message("<span class='warning'>[user] flips \the [src]!</span>")
+
+
+		user.visible_message("<span class='warning'>[user] flips [src]!</span>")
 
 		if(climbable)
 			structure_shaken()
 	else
+		if(flip_speed > 0)
+			user.visible_message("<span class='warning'[user] starts trying to right [src]!</span>", "<span class='warning'>You start trying to right [src][flip_speed >= 5 SECONDS ? " (it'll take about [flip_speed / 10] seconds)." : ""]</span>")
+			if(!do_after(user, flip_speed, TRUE, src))
+				user.visible_message("<span class='notice'[user] gives up on trying to right [src].</span>")
+				return
 		if(!unflip())
 			to_chat(user, "<span class='notice'>It won't budge.</span>")
+		user.visible_message("<span class='warning'>[user] rights [src]!</span>")
+
+
+/obj/structure/table/proc/get_flip_speed(mob/living/flipper)
+	if(!istype(flipper))
+		return 0  // sure
```

Comment:
Can you update the comment here to show that this is not actually returning `FALSE`? It might seem obvious from the other returns, but it might be a good idea regardless.